Quelch Warnings

mxmlnkn opened the pull request:
This squelches some warnings in debug mode which distract when trying to find problems in new projects. Instead of deleting variable names on unused params warnings, I also tend to comment them out with /**/ if the name adds an insightful description. Also the CMakeLists.txt file seem to still contain trailing spaces, I fogot them in my find command in one of my earlier pull requests, they got auto deleted by my editor.
Edit: yes, I failed to spell 'squelch' correctly in the commit messages.

mxmlnkn committed:
quelch compiler warnings for Wsigned-compare

mxmlnkn committed:
return this reference for copy constructor

mxmlnkn committed:
quelch Wreorder warnings

mxmlnkn committed:
add -Wextra

mxmlnkn committed:
quelch -Wignored-qualifiers warnings

mxmlnkn committed:
quelch unused param warnings

mxmlnkn committed:
add correct lemonade compile dependency for other two examples

Commit:
Merge pull request #83 from mxmlnkn/makej

make parallel compilation with make -j on first compilation work

MartinWenge committed:
Merge branch 'warnings' of https://github.com/mxmlnkn/LeMonADE into develop

MartinWenge commented:
in SlowBondset.cpp: 
- In copy constructor of slow bondset: add default constructor of FastBondset 
- remove explicit assingment of lookupSynchronized=false

in ConnectedDecorator.h:
- delete the debug output std::cerr<< "Connected::connect(unit b): Vertices already connected" << std::endl; in line 220

mxmlnkn commented:
"Allow edits from maintainers" is checked (by default, it seems)

MartinWenge committed:
in SlowBondset.cpp:

    In copy constructor of slow bondset: add default constructor of FastBondset
    remove explicit assingment of lookupSynchronized=false

in ConnectedDecorator.h:

    delete the debug output std::cerr<< "Connected::connect(unit b): Vertices already connected" << std::endl; in line 220
